xtask/src/cli.rs

```diff
@@ -0,0 +1,60 @@
+use std::process::exit;
+
+use anyhow::{anyhow, bail, Context};
+use pico_args::Arguments;
+
+const HELP: &str = "\
+Usage: xtask <COMMAND>
+
+Commands:
+  run-wasm
+
+Options:
+  -h, --help  Print help
+";
+
+pub(crate) struct Args {
+    pub(crate) subcommand: Subcommand,
+}
+
+impl Args {
+    pub fn parse() -> Self {
+        let mut args = Arguments::from_env();
+        log::debug!("parsing args: {args:?}");
+        if args.contains("--help") {
+            eprint!("{HELP}");
+            exit(101);
+        }
+        match (|| -> anyhow::Result<_> {
+            let subcommand = Subcommand::parse(args)?;
+            Ok(Self { subcommand })
+        })() {
```

Comment:
Immediately invoked lambdas look really weird - what's the purpose for them in this case? Maybe a named function would work better.